service/windows/installer/custom-actions/src/lib.rs

```diff
@@ -0,0 +1,173 @@
+use std::ffi::{OsStr, OsString};
+use std::fs::remove_dir_all;
+use std::os::windows::ffi::{OsStrExt, OsStringExt};
+use std::path::Path;
+use std::ptr;
+use winapi::shared::minwindef::{DWORD, LPDWORD, UINT};
+use winapi::shared::winerror::{ERROR_FUNCTION_NOT_CALLED, ERROR_MORE_DATA, ERROR_SUCCESS};
+use winapi::um::winnt::{LPCWSTR, LPWSTR};
+use winapi::um::winsvc::*;
+
+type MsiHandle = u32;
+
+extern "system" {
+    fn MsiGetPropertyW(
+        hInstall: MsiHandle,
+        szName: LPCWSTR,
+        szValueBuf: LPWSTR,
+        pcchValueBuf: LPDWORD,
+    ) -> UINT;
+}
+
+/// A custom action to delete a directory and all of its contents.
+#[no_mangle]
+pub extern "system" fn DeleteDataFolder(install_handle: MsiHandle) -> UINT {
```

Comment:
Generally the recommendation and style we try to use is that `extern` functions are not public.

Reply:
Is this documented somewhere? Generally, it seems pretty common (including in our code) to use `pub extern`, and this is what is used in the [rust documentation](https://doc.rust-lang.org/std/keyword.extern.html) also.